contracts/factory/src/contract.rs

```diff
@@ -501,6 +438,56 @@ impl FactoryTrait for Factory {
 
 #[contractimpl]
 impl Factory {
+    #[allow(clippy::too_many_arguments)]
+    pub fn __constructor(
+        env: Env,
+        admin: Address,
+        multihop_wasm_hash: BytesN<32>,
+        lp_wasm_hash: BytesN<32>,
+        stable_wasm_hash: BytesN<32>,
+        stake_wasm_hash: BytesN<32>,
+        token_wasm_hash: BytesN<32>,
+        whitelisted_accounts: Vec<Address>,
+        lp_token_decimals: u32,
+    ) {
+        if is_initialized(&env) {
+            log!(
+                &env,
+                "Factory: Initialize: initializing contract twice is not allowed"
+            );
+            panic_with_error!(&env, ContractError::AlreadyInitialized);
+        }
```

Comment:
Aren't these redundant now?

Reply:
not 100% sure they are - during testing, we can still get to this error if we set the same address.